opa/opa.go

```diff
@@ -0,0 +1,212 @@
+package opa
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/server/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/observatorium/observatorium/rbac"
+)
+
+// FailMode defines the default failure mode for an authorizer, i.e. whether queries should fail open or closed.
```

Comment:
For a circuit to fail open or closed means what should be the default when there is a failure. The authorizers based on OPA can error on where the standard rbac authorizer won't, so it can be useful to set the defaul behavior if there is a problem, e.g. contacting the OPA server. If a user knows what they are doing but breaks something in their server's configuration, which causes an authorizer to fail, then they can fail open so as to not lock themselves out. We can also simply always return `false` in the case of errors and let demand for the feature drive its development at a later time.